Dockerfile

```diff
@@ -0,0 +1,203 @@
+RUN echo "build --spawn_strategy=standalone --genrule_strategy=standalone" \
+    >>/etc/bazel.bazelrc
+
+# Put cuda libraries to where they are expected to be
+RUN ln -s /usr/local/cuda/lib64/stubs/libcuda.so /usr/local/cuda/lib64/stubs/libcuda.so.1
+RUN cp /usr/include/cudnn.h /usr/local/cuda/include/cudnn.h
+
+# Set library paths
+ENV LD_LIBRARY_PATH $LD_LIBRARY_PATH:/usr/local/cuda/extras/CUPTI/lib64:/usr/local/cuda/lib64:/usr/lib/x86_64-linux-gnu/:/usr/local/cuda/lib64/stubs/
+
+# << END Configure Bazel
+
+
+
+
+# Copy DeepSpeech repo contents to container's /DeepSpeech
+COPY . /DeepSpeech/
+
+WORKDIR /DeepSpeech
+
+RUN pip --no-cache-dir install -r requirements.txt
+
+# Link DeepSpeech native_client libs to tf folder
+RUN ln -s /DeepSpeech/native_client /tensorflow
+
+
+
+
+# >> START Build and bind
+
+WORKDIR /tensorflow
+
+# passing LD_LIBRARY_PATH is required cause Bazel doesn't pickup it from environment
+
+# Build LM Prefix Decoder
+RUN bazel build -c opt --copt=-O3 --config=cuda --copt="-D_GLIBCXX_USE_CXX11_ABI=0" //native_client:libctc_decoder_with_kenlm.so  --verbose_failures --action_env=LD_LIBRARY_PATH=${LD_LIBRARY_PATH}
+
+# Build DeepSpeech
+RUN bazel build --config=monolithic -c opt --copt=-O3 --copt="-D_GLIBCXX_USE_CXX11_ABI=0" --copt=-fvisibility=hidden //native_client:libdeepspeech.so //native_client:deepspeech_utils //native_client:generate_trie --verbose_failures --action_env=LD_LIBRARY_PATH=${LD_LIBRARY_PATH}
+
+
+# Build TF pip package
+RUN bazel build --config=opt --config=cuda  --copt=-msse4.2 //tensorflow/tools/pip_package:build_pip_package --verbose_failures --action_env=LD_LIBRARY_PATH=${LD_LIBRARY_PATH}
+
+# Fix for not found script https://github.com/tensorflow/tensorflow/issues/471
+RUN ./configure
+
+# Build wheel
+RUN bazel-bin/tensorflow/tools/pip_package/build_pip_package /tmp/tensorflow_pkg
+
+# Install tensorflow from our custom wheel
+RUN pip install /tmp/tensorflow_pkg/*.whl
+
+# Copy built libs to /DeepSpeech/native_client
+RUN cp /tensorflow/bazel-bin/native_client/libctc_decoder_with_kenlm.so /DeepSpeech/native_client/ \
+    && cp /tensorflow/bazel-bin/native_client/generate_trie /DeepSpeech/native_client/ \
+    && cp /tensorflow/bazel-bin/native_client/libdeepspeech.so /DeepSpeech/native_client/ \
+    && cp /tensorflow/bazel-bin/native_client/libdeepspeech_utils.so /DeepSpeech/native_client/
+ 
+
+# Make DeepSpeech and install Python bindings
+ENV TFDIR /tensorflow
+WORKDIR /DeepSpeech/native_client
+RUN make deepspeech
+RUN make bindings
+RUN pip install dist/deepspeech*
+
+
+# << END Build and bind
+
+
+
+
+# Allow Python printing utf-8
+ENV PYTHONIOENCODING UTF-8
+
+# Build KenLM in /DeepSpeech/native_client/kenlm folder
+WORKDIR /DeepSpeech/native_client
+RUN rm -rf kenlm \
+    && git clone https://github.com/kpu/kenlm && cd kenlm \
+    && mkdir -p eigen3 \
+    && cd eigen3 && wget -O - https://bitbucket.org/eigen/eigen/get/3.2.8.tar.bz2 |tar xj && cd - \
```

Comment:
Is Eigen a new dependency for KenLM? I've never needed it to create language models.

Reply:
Actually its required only for interpolation feature and not needed for building LM. Successfully building without Eigen.